lib/auth/userloginstate/generator.go

```diff
@@ -186,19 +191,31 @@ func (g *Generator) addAccessListsToState(ctx context.Context, user types.User,
 	}
 
 	for _, accessList := range accessLists {
-		if err := services.IsAccessListOwner(identity, accessList); err == nil {
-			g.grantRolesAndTraits(identity, accessList.Spec.OwnerGrants, state)
+		if dynamic, err := services.IsAccessListOwner(ctx, g.accessLists, identity, accessList); err == nil {
```

Comment:
What is dynamic ? Can we rename it to isOwner ? 

Also you can simplify the logic by checking: 
```go
if isOwner, err := services.IsAccessListOwner(ctx, g.accessLists, identity, accessList); err == nil &&  {
   added := g.grantRolesAndTraits(identity, accessList.Spec.OwnerGrants, state)
   inheritedGrants.Roles = append(inheritedGrants.Roles, added.Roles...)
  for k, values := range added.Traits {
	  inheritedGrants.Traits[k] = append(inheritedGrants.Traits[k], values...)
  }
}
````